step-maven-plugin/step-commons-maven-plugin/src/main/java/step/plugins/maven/AbstractStepPluginMojo.java

```diff
@@ -86,7 +103,53 @@ protected ControllerCredentials getControllerCredentials() {
 		return new ControllerCredentials(getUrl(), null);
 	}
 
-	protected Artifact getArtifactByClassifier(String artifactClassifier, String groupId, String artifactId, String artifactVersion) {
+	protected String resolveKeywordLibResourceByCriteria(Map<String, String> libStepResourceSearchCriteria) throws MojoExecutionException {
+		getLog().info("Using Step resource " + libStepResourceSearchCriteria + " as library file");
+
+		if (libStepResourceSearchCriteria.containsKey("id")) {
+			// just use the specified id
+			return libStepResourceSearchCriteria.get("id");
+		} else {
+			// search resources by attributes except for id
+			Map<String, String> attributes = new HashMap<>(libStepResourceSearchCriteria);
+			attributes.remove("id");
+			AbstractAccessor<Resource> remoteResourcesAccessor = createRemoteResourcesAccessor();
+			List<Resource> foundResources = StreamSupport.stream(remoteResourcesAccessor.findManyByAttributes(attributes), false).collect(Collectors.toList());
+			if (foundResources.isEmpty()) {
+				throw new MojoExecutionException("Library resource is not resolved by attributes: " + attributes);
+			} else if (foundResources.size() > 1) {
+				throw new MojoExecutionException("Ambiguous library resources ( " + foundResources.stream().map(AbstractIdentifiableObject::getId).collect(Collectors.toList()) + " ) are resolved by attributes: " + attributes);
+			} else {
+				return foundResources.get(0).getId().toString();
+			}
+		}
+	}
+
+	protected AbstractAccessor<Resource> createRemoteResourcesAccessor() {
+		RemoteAccessors remoteAccessors = new RemoteAccessors(new RemoteCollectionFactory(getControllerCredentials()));
```

Comment:
@iegorov777 Has the tenant to be specified somewhere?

Reply:
@jeromecomte tenant is resolved in child classes for EE (RunPackagedAutomationPackagesMojoEE,  RunDeployedAutomationPackagesMojoEE), where we create  the `TenantSwitcher` to switch current tenant to the `stepProjectName` property.